src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java

```diff
@@ -253,8 +253,10 @@ private int getTotalRecords(JsonObject records) {
   }
 
   private PatronInformationResponseBuilder addPersonalData(Personal personal,
-      PatronInformationResponseBuilder builder) {
-    final String personalName = getPatronPersonalName(personal);
+                                                     String patronIdentifier,
+                                                     PatronInformationResponseBuilder builder) {
+    String personalName = getPatronPersonalName(personal);
+    personalName = personalName.isEmpty() ? patronIdentifier : personalName;
```

Comment:
Maybe we could move this logic into `getPatronPersonalName` so that, in case we need to get the personal name somewhere else in the future, we don't need to repeat this logic.

In there, we could use `Collectors.collectingAndThen` to check if the string returned is empty and, if it is, return a "default" name. Something like:

```Java
  private String getPatronPersonalName(Personal personal, String defaultPersonalName) {
    if (personal != null) {
      return Stream.of(personal.getFirstName(), personal.getMiddleName(), personal.getLastName())
          .filter(Objects::nonNull)
          .collect(Collectors.collectingAndThen(Collectors.joining(" "), s -> s.isEmpty() ? defaultPersonalName : s));
    }

    return defaultPersonalName;
  }
```

Reply:
Cool. I wanted to add this logic there but didn't know of a way to poke into this lambda expression.

Reply:
I hope it works as is, I didn't compile it, but I've tried code like this in the past. It passed the eyeball test. 😄